src/etaoin/api.clj

```diff
@@ -3546,9 +3547,29 @@
                          (drv/set-capabilities (:capabilities defaults-global))
                          (drv/set-capabilities (get-in defaults [type :capabilities]))
                          (drv/set-capabilities capabilities)))
-        caps          (:capabilities driver)
-        session       (create-session driver caps)]
-    (assoc driver :session session)))
+        caps          (:capabilities driver)]
+    (loop [n 1]
+      ;; Wait for driver to be ready before creating the first session.
+      (wait-predicate
+       (fn [] (-> (get-status driver) :ready))
+       {:timeout 30
+        :interval 0.100
+        :message "Timeout waiting for WebDriver to become ready after creation."})
+      (let [driver (assoc driver :session (create-session driver caps))
+            good-session (try (get-window-handle driver)
+                              true
+                              (catch Exception e
+                                (prn e)
+                                false))
+            ]
+        (if good-session
+          driver
+          (do 
+            (delete-session driver)
+            (if (< n 3)                 ; max attempts = 3
+              (recur (inc n))
+              (throw+ {:type :etaoin/retries
```

Comment:
Could save ex above and throw it as cause.

Reply:
Yes. Good catch.

Reply:
Done.